server/api.ts

```diff
@@ -1104,6 +1105,33 @@ router.postAsync("/stage-stats", ensure.isAdmin, async (req, res) => {
   res.send(updatedStageStats);
 });
 
+/**
+ * POST /api/refetch-stats
+ * Refretches the stats of a batch of mp links in a stage
+ * Params:
+ *   - tourney: identifier for the tourney
+ *   - stage: identifier for the stage
+ * Returns:
+ *   - stats: the updated stage stats
+ */
+router.postAsync("/refetch-stats", ensure.isAdmin, async (req, res) => {
+  logger.info(`${req.user.username} initiated a refresh of ${req.body.tourney}/${req.body.stage} stats`);
+  let mpLinks: string[] = [];
+  if (req.body.stage === "Qualifiers") mpLinks = (await QualifiersLobby.find({ tourney: req.query.tourney })).map(lobby => lobby.link);
+  else mpLinks = (await Match.find({ tourney: req.body.tourney, stage: req.body.stage })).map(match => match.link);
+  for (const mpLink of mpLinks) {
+    if (mpLink) {
+      const found = mpLink.match(mpRegex1) || mpLink.match(mpRegex2);
+      if (found) await fetchMatchAndUpdateStageStats(req.body.tourney, req.body.stage, found[1]);
```

Comment:
hm this has lots of repeated work
for every match, `fetchMatchAndUpdateStageStats` refetches every map in the stage and every player and every team in the tournament which is kinda wild
would be nice to change this, but i guess if it's a pain then maybe not necessary rn

Reply:
true... in that case, I will modify the function so it can take a list of match IDs